src/pspm_cfg/pspm_cfg_pp_scr.m

```diff
@@ -1,249 +1,231 @@
+chan_action.val       = {'replace'};
+chan_action.help      = {['Choose whether to add the new channels or ',...
+                          'replace a channel previously added by ',...
+                          'this method.']};
+% Executable Branch
+pp_scr                = cfg_exbranch;
+pp_scr.name           = 'Preprocessing SCR';
+pp_scr.tag            = 'pp_scr';
+pp_scr.val            = {datafile, ...
+                         scr_min, ...
+                         scr_max, ...
+                         scr_slope, ...
+                         scr_def_thr, ...
+                         scr_isl_thr, ...
+                         scr_exp_ep, ...
+                         chan, ...
+                         clip_det, ...
+                         missepoch, ...
+                         chan_action};
+pp_scr.prog           = @pspm_cfg_run_scr_pp;
+pp_scr.vout           = @pspm_cfg_vout_scr_pp;
+pp_scr.help           = {['Pre processing (PP) skin conductance ',...
+                          'response (SCR).'],...
+                         ['See I. R. Kleckner et al., "Simple, ' ...
+                          'Transparent, and Flexible Automated ' ...
+                          'Quality Assessment Procedures for ' ...
+                          'Ambulatory Electrodermal Activity Data," ' ...
+                          'in IEEE Transactions on Biomedical ' ...
+                          'Engineering, vol. 65, no. 7, ' ...
+                          'pp. 1460--1467, July 2018.']};
+  function vout = pspm_cfg_vout_scr_pp(~)
+    vout = cfg_dep;
+    vout.sname      = 'Output Channel';
+    vout.tgt_spec = cfg_findspec({{'class','cfg_entry'}});
+    vout.src_output = substruct('()',{':'});
+  end
+  function out = pspm_cfg_run_scr_pp(job)
+    scr_pp_datafile = job.datafile{1};
+    scr_pp_options = struct();
+    scr_pp_options.min = job.min;
+    scr_pp_options.max = job.max;
+    scr_pp_options.slope = job.slope;
+    scr_pp_options.deflection_threshold = job.deflection_threshold;
+    scr_pp_options.expand_epochs = job.expand_epochs;
+    scr_pp_options.channel = pspm_cfg_channel_selector('run', job.chan);
+    scr_pp_options.clipping_step_size = job.clipping_detection.clipping_step_size;
+    scr_pp_options.clipping_threshold = job.clipping_detection.clipping_threshold;
+    if isfield(job.missing_epochs, 'write_to_file')
+      scr_pp_options.missing_epochs_filename = [job.missing_epochs.write_to_file.outdir{1},...
+        '/', job.missing_epochs.write_to_file.filename, '.mat'];
+      length_temp = length(scr_pp_options.missing_epochs_filename);
+      if length_temp > 7
+        if strcmp((scr_pp_options.missing_epochs_filename(length_temp-7: length_temp)),'.mat.mat')
+          scr_pp_options.missing_epochs_filename(length_temp-3:length_temp) = [];
         end
+      end
+    end
+    scr_pp_options.channel_action = job.chan_action;
+    [sts, output] = pspm_scr_pp(scr_pp_datafile, scr_pp_options, scr_pp_options.channel);
```

Comment:
I correct this file by adding `scr_pp_options.channel`. In `pspm_scr_pp`, `channel` is an independent variable from `options`.